trinity/initialization.py

```diff
@@ -106,8 +106,14 @@ def initialize_database(chain_config: ChainConfig, chaindb: AsyncChainDB, base_d
         chaindb.get_canonical_head()
     except CanonicalHeadNotFound:
         chain_class = chain_config.full_chain_class
+        genesis_params = chain_config.genesis_params.to_dict()
+        genesis_state = {
+            address: account.to_dict()
+            for address, account
+            in chain_config.genesis_state.items()
+        }
         chain_class.from_genesis(
             base_db=base_db,
-            genesis_params=chain_config.genesis_params,
-            genesis_state=chain_config.genesis_state,
+            genesis_params=genesis_params,
+            genesis_state=genesis_state,
```

Comment:
Maybe some of this logic that is just dumping in data from `chain_config` could be pushed inside `chain_config`, so the API could be:

```py
chain = chain_config.launch_genesis(base_db)
```

If you like it